pkg/virt-controller/watch/application.go

```diff
@@ -658,6 +661,10 @@ func (vca *VirtControllerApp) initCommon() {
 		topologyHinter,
 		netAnnotationsGenerator,
 		netvmicontroller.UpdateStatus,
+		func(field *k8sfield.Path, vmiSpec *v1.VirtualMachineInstanceSpec, clusterCfg *virtconfig.ClusterConfig) []metav1.StatusCause {
+			networkValidator := netadmitter.NewValidator(field, vmiSpec, clusterCfg)
+			return networkValidator.ValidateCreation()
+		},
```

Comment:
Its possible, but these is a con: it introduce a dependency of `pkg/network/admitters` on `pkg/virt-config`, because the mentioned function signature (admitters.ValidateSpec) specifics `*virtconfig.ClusterConfig`.
WDYT?